lib/charms/layer/docker_registry.py

```diff
@@ -143,6 +142,8 @@ def configure_registry():
         storage['delete'] = {'enabled': True}
     if charm_config.get('storage-read-only'):
         storage['maintenance'] = {'readonly': {'enabled': True}}
+    storage_cache = charm_config.get('storage-cache', 'inmemory')
+    storage['cache'] = {'blobdescriptor': storage_cache}
```

Comment:
Also, should we validate that these are the only two valid options (e.g., `inmemory`, `disabled`) and remove the cache structure if an unknown value is set? The [docs](https://distribution.github.io/distribution/about/configuration/#cache) suggest we use it if needed and omit it otherwise.

Reply:
yup, good call.  i've updated configure and unit test so we only populate a [storage][cache] structure if 'inmemory' is specified.  anything else omits it.